FWCore/Framework/src/ProductResolvers.cc

```diff
@@ -1109,7 +1103,7 @@ namespace edm {
                                                           SharedResourcesAcquirer* sra,
                                                           ModuleCallingContext const* mcc,
                                                           ServiceToken token,
-                                                          oneapi::tbb::task_group* group) const {
+                                                          oneapi::tbb::task_group* group) const noexcept {
```

Comment:
We should inspect `Principal:: getProductResolverByIndex()` (probably just mark it as noexcept)